packages/data-point-express/lib/middleware.js

```diff
@@ -42,20 +42,23 @@ function createErrorMessage(err) {
   };
 }
 
-function resolveReducer(dataPoint, reducer, options, res, initialValue = {}) {
-  dataPoint
-    .transform(reducer, initialValue, options)
-    .then(acc => {
-      if (typeof acc.value === "string") {
-        res.send(acc.value);
-        return;
-      }
-      res.json(acc.value);
-    })
-    .catch(err => {
-      res.status(400).send(createErrorMessage(err));
-    })
-    .done();
+async function resolveReducer(
+  dataPoint,
+  reducer,
+  options,
+  res,
+  initialValue = {}
+) {
+  try {
+    const acc = await dataPoint.transform(reducer, initialValue, options);
+    if (typeof acc.value === "string") {
+      res.send(acc.value);
+      return;
```

Comment:
Do you need to return here? Is it possible to use if else instead of return?

Reply:
we do want to exit as soon as the response is sent, yes an else could also be used, its a matter of "style" at this point. 

I usually do returns after an early response because I've had race conditions before or just unexpected code being executed and had to spend some time trying to figure out where or why this was happening. 

In the end, I believe it comes down to personal preference, both are as valid as long as the logic of the code follows expectations.